pytta/classes/signal.py

```diff
@@ -260,24 +280,29 @@ def freqSignal(self, newSignal):
                 newSignal = np.array(newSignal, ndmin=2)
             if newSignal.shape[1] > newSignal.shape[0]:
                 newSignal = newSignal.T
-            self._freqSignal = np.array(newSignal, dtype='complex64')
-            # [-] number of samples
-            # needed when freqSignal is provided at init
-            if hasattr(self, '_timeSignal'):
-                self._numSamples = len(self.timeSignal)
+            # Time numSamples was provided (or is default) at init or old
+            # signal was already here. Check if numSamples calculated
+            # according to half spectra matches the current numSamples.
+            halfSpectraNumSamples = len(newSignal)
+            if (halfSpectraNumSamples-1)*2 == self._numSamples:
+                timeSignalNumSamplesIs = "EVEN"
+            elif halfSpectraNumSamples*2-1 == self._numSamples:
+                timeSignalNumSamplesIs = "ODD"
             else:
-                freqLen = len(self.freqSignal)
-                self._numSamples = int((freqLen-1)*2 + 1) \
-                                   if freqLen % 2 == 0 \
-                                   else int((freqLen*2) - 2)
+                # Old numSamples don't match with provided half spectrum
+                # number of samples.
+                timeSignalNumSamplesIs = "UNKNOWN"  
+                
+            if timeSignalNumSamplesIs == "UNKNOWN":
+                # Consider full spectrum has even number of samples
+                # self._numSamples = halfSpectraNumSamples*2-1  # ODD
```

Comment:
All these lines could be removed, then the `else` statement will directly assume an "EVEN" time signal. It could be the case that only the checking for the "ODD" is made, since every other case will fall down to assume an even signal.

Reply:
You are right. I did this way just to extremely clear what is happening. Maybe It was too much :P